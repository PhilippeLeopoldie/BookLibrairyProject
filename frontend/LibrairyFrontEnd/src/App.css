#root {
  margin: 0 auto;
  text-align: center;
}

.nav {
  font-size: x-large;
}

.nav__element {
  color: blue;
}

.nav__element:hover {
  font-size: xx-large;
}
