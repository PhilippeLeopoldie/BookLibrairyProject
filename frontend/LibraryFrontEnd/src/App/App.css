#root {
  text-align: center;
}

.App {
  display: flex;
  flex-direction: column;
}

.App_navBar--container {
    position: relative;
    background-color: inherit;
}



.Contents {
    background-color: inherit;
    margin: 2rem 0rem 2rem 0rem;
}

.App--black{
  background-color: var(--backgroundcolor-darkmode);
  .navbar--black {
    background-color:inherit;
  }
}

@media screen and (min-width: 700px) {
    .Contents {
        margin: 5rem 0rem 0rem 0rem;
    } 
}
