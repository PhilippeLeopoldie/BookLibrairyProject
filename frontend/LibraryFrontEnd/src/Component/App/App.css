#root {
  text-align: center;
}

.App {
  display: flex;
  flex-direction: column;
}

.App--black{
  background-color: #000000;
  .navbar--black{
    background-color:#000000;
  }
}
