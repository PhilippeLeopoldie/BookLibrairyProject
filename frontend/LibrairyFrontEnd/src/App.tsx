import { useContext } from "react";
import { Routes, BrowserRouter, Route, Link } from "react-router-dom";
import { OpinionContext } from "./Context";
import "./App.css";
import AddBook from "./Component/AddBook";
import OpinionUpdate from "./Component/OpinionUpdate";
import Home from "./Component/Home";

function App() {
  const initialContext = useContext(OpinionContext);

  return (
    <div className="App">
      
      <BrowserRouter>
      <div className="nav">
      <ul className="liste">
          <dl className="nav-data">
            <Link className="nav-link home" to="/">Home</Link>
          </dl>
          <dl className="nav-data">
            <Link  className="nav-link" to="/addBook">Add a review</Link>
          </dl>
        </ul>

      </div>
        
        <Routes>
          <Route path="/" element={<Home />}></Route>
          <Route path="/addBook" element={<AddBook />}></Route>
          <Route
            path="/viewUpdate"
            element={
              <div>
                <OpinionContext.Provider value={initialContext}>
                  <OpinionUpdate />
                </OpinionContext.Provider>
              </div>
            }
          ></Route>
        </Routes>
      </BrowserRouter>

      
      
    </div>
  );
}
export default App;
