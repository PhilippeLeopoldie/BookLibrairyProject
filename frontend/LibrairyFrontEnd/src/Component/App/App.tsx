import { useContext } from "react";
import { Routes, BrowserRouter, Route, Link } from "react-router-dom";
import { OpinionContext } from "../../Context";
import "./App.css";
import {AddBook} from "../AddBook/AddBook";
import {OpinionEdit} from "../OpinionEdit/OpinionEdit";

import { BookList } from "../BookList/BookList";

export const App = () => {
  const initialContext = useContext(OpinionContext);

  return (
    <div className="App"> 
      <BrowserRouter>
      <div className="nav">
        <ul className="liste">
          <dl className="nav-data">
            <Link className="nav-link home" to="/">Home</Link>
          </dl>
          <dl className="nav-data">
            <Link  className="nav-link" to="/addBook">Add a review</Link>
          </dl>
        </ul>
      </div>
        
        <Routes>
          <Route path="/" element={<BookList />}></Route>
          <Route path="/addBook" element={<AddBook />}></Route>
        </Routes>
      </BrowserRouter>
    </div>
  );
}
