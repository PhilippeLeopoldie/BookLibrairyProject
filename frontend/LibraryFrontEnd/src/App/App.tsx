import "./App.css";
import { About } from "../Component/About/About";
import { BookCreate } from "../Component/BookCreate/BookCreate";
import { GENRES_LIST_URL } from "../constants/api";
import { HomePage } from "../Component/HomePage/HomePage";
import { Routes, BrowserRouter, Route } from "react-router-dom";
import { BookSearch } from "../Component/BookSearch/BookSearch";
import { createContext, useContext, useEffect, useState } from "react";
import { NavBar } from "../Component/NavBar/NavBar";
import { SideBar } from "../Component/SideBar/SideBar"
import { StoryCard } from "../Component/Story/StoryCard";


type ButtonContextType = {
    buttonStatus: string,
    setButtonStatus: (newStatus: string) => void;
}

type BookType = {
    id: number;
    title: string;
    author: string;
    imageUrl?: string;
    averageRate: number;
    opinions?: OpinionType | null;
};

type OpinionType = {
    rate: number;
    view: string;
    userName: string;
};

type GenreResponseType = {
    genres: GenreType[],
    totalGenreCount: number,
}

type GenreType = {
    id: number,
    name: string,
    books: BookType[]
}

type genresCacheContextType = {
    genresCache: GenreResponseType | null,
    setGenresCache: (newStatus: GenreResponseType) => void
}

type NewBooksCacheContextType = {
    newBooksCache: BookType[] | null,
    setNewBooksCache: (newStatus: BookType[]) => void
}

type TopBooksCacheContextType = {
    topBooksCache: BookType[] | null,
    setTopBooksCache: (newStatus: BookType[]) => void
}

export const ThemeContext = createContext<string>("black");
export const ButtonContext = createContext<ButtonContextType | undefined>(undefined);
export const newBooksCacheContext = createContext<NewBooksCacheContextType | null>(null);
export const topBooksCacheContext = createContext<TopBooksCacheContextType | null>(null);
export const genresCacheContext = createContext<genresCacheContextType | null>(null);

export const App = () => {
    const rootElement = document.documentElement;
    const [theme, setTheme] = useState<string>("black");
    const [buttonStatus, setButtonStatus] = useState<string>("inactivated");
    const [newBooksCache, setNewBooksCache] = useState<BookType[] | null>(null);
    const [topBooksCache, setTopBooksCache] = useState<BookType[] | null>(null);
    const [genresCache, setGenresCache] = useState<GenreResponseType | null>(null);

    const handleTheme = () => {
        setTheme(theme === "natural" ? "black" : "natural");
    }

    const fetchGenres = async () => {
        try {
            const genreResponse: Response = await fetch(GENRES_LIST_URL);
            if (genreResponse.status === 200) {
                const genreResponseData: GenreResponseType = await genreResponse.json();
                setGenresCache(genreResponseData);
                console.log(genreResponseData.genres);
            } else if (genreResponse.status === 404) {
                console.log(genreResponse);
            }
        } catch (error) {
            console.error("Error fetching genres", error);
        }
    }
    useEffect(() => {
        if (!genresCache) fetchGenres();
    }, []);
    
    return (
        <div className={`App App--${theme}`}>
            <script>
                {theme === "black"
                    ? (rootElement.style.backgroundColor = "#1e1e1e")
                    : (rootElement.style.backgroundColor = "#F5F5F5")}
            </script>
            <ThemeContext.Provider value={theme}>
                <ButtonContext.Provider value={{ buttonStatus, setButtonStatus }}>
                    <newBooksCacheContext.Provider value={{ newBooksCache, setNewBooksCache }}>
                        <topBooksCacheContext.Provider value={{ topBooksCache, setTopBooksCache }}>
                            <genresCacheContext.Provider value={{ genresCache, setGenresCache }}>
                            <BrowserRouter>
                                <div className={`App App--${theme}`}>
                                    <section className={`App--${theme} App_navBar--container`}>
                                        <NavBar theme={theme} handleTheme={handleTheme} />
                                        <SideBar theme={theme} handleTheme={handleTheme} />
                                    </section>
                                    <section className="Contents">
                                        <Routes>
                                            <Route path="/" element={<HomePage />} />
                                            <Route path="/about" element={<About/>} />
                                            <Route path="/addBook" element={<BookCreate />} />
                                            <Route path="/Search" element={<BookSearch />} />
                                            <Route path="/createStory" element={<StoryCard />} />
                                        </Routes>
                                    </section>
                                </div>
                                </BrowserRouter>
                            </genresCacheContext.Provider>
                        </topBooksCacheContext.Provider>
                    </newBooksCacheContext.Provider>
                </ButtonContext.Provider>
            </ThemeContext.Provider>
        </div>
    );
};
