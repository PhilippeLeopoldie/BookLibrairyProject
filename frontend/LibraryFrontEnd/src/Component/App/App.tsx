import "./App.css";
import { BookCreate } from "../BookCreate/BookCreate";
import { BookList } from "../BookList/BookList";
import { Routes, BrowserRouter, Route, Link } from "react-router-dom";
import { BookSearch } from "../BookSearch/BookSearch";
import { createContext, useContext, useState } from "react";
import addIcone from "../../media/add20px.svg";
import bookIcone from "../../media/book48px.svg";
import homeIcone from "../../media/home_48px.svg";
import searchIcone from "../../media/search48px.svg";
import settings from "../../media/settings48px.svg";

export const ThemeContext = createContext<string | null>(null);

export const App = () => {
  const [theme, setTheme] = useState<string>("natural");

  return (
    <div className="App">
      <ThemeContext.Provider value={theme}>
        <BrowserRouter>
          <nav className={"navbar--grid navbar--" + theme}>
            <Link className="nav-link nav__home__Link--flex" to="/">
              <img
                className="nav-link icone nav__HomeIcone"
                src={homeIcone}
                alt="Home"
              ></img>
              <figcaption className="iconeTitle nav__Home__title">
                Home
              </figcaption>
            </Link>
            <Link className="nav-link nav__search__link--flex" to="/Search">
              <img
                className="nav-link icone nav__search__image"
                src={searchIcone}
                alt="Search"
              />
              <figcaption className="iconeTitle nav__search__title">
                Search
              </figcaption>
            </Link>
            <Link className="nav-link nav__addBook--flex" to="/addBook">
              <img
                className="nav-link icone nav__addBook"
                src={bookIcone}
                alt="book"
              />
              <img className="nav-link addIcone" src={addIcone} alt="add" />
              <figcaption className="iconeTitle nav__AddBook__Title">
                Add a book
              </figcaption>
            </Link>
            <div className="nav-link icone settingsIcone">
              <img
                className="nav-link"
                src={settings}
                onClick={() => {
                  setTheme(theme === "natural" ? "antique" : "natural");
                }}
                alt="Themes"
              />
              <figcaption className="iconeTitle nav__Settings__Title">Themes</figcaption>
            </div>
          </nav>
          <Routes>
            <Route path="/" element={<BookList />} />
            <Route path="/addBook" element={<BookCreate />} />
            <Route path="/Search" element={<BookSearch />} />
          </Routes>
        </BrowserRouter>
      </ThemeContext.Provider>
    </div>
  );
};
