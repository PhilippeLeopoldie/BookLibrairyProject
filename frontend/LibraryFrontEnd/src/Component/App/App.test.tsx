import {render, screen,} from "@testing-library/react";
import { App } from "./App";
import "@testing-library/jest-dom";

describe("App routes", () => {
  test("snapShot: render nav bar", ()=> {
    const {asFragment} = render(<App/>)

    expect(asFragment()).toMatchSnapshot();
  })
})