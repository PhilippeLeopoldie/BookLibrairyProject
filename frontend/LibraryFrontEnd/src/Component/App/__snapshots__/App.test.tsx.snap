// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App routes snapShot: render nav bar 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="navbar"
    >
      <dl
        class="navElement navbar__home"
      >
        <a
          class="nav-link homeLink"
          href="/"
        >
          <img
            alt="Home"
            class="nav-link nav__HomeIcone"
            src="SvgrURL"
          />
          <figcaption
            class="iconeTitle nav__Home__title"
          >
            Home
          </figcaption>
        </a>
      </dl>
      <dl
        class="navElement navbar__search"
      >
        <a
          class="nav-link"
          href="/Search"
        >
          <img
            alt="Search"
            class="nav-link nav__search"
            height="48"
            src="SvgrURL"
            width="48"
          />
          <figcaption
            class="iconeTitle nav__search__title"
          >
            Search
          </figcaption>
        </a>
      </dl>
      <dl
        class="navElement navbar__addBook"
      >
        <a
          class="nav-link addBook"
          href="/addBook"
        >
          <img
            alt="book"
            class="nav-link nav__addBook"
            height="48"
            src="SvgrURL"
            width="48"
          />
          <img
            alt="add"
            class="nav-link addIcone"
            height="20"
            src="SvgrURL"
            width="20"
          />
          <figcaption
            class="iconeTitle nav__AddBook__Title"
          >
            Add a book
          </figcaption>
        </a>
      </dl>
    </div>
    <h1>
      Loading...
    </h1>
  </div>
</DocumentFragment>
`;
